### ISSUE TITLE
some stream nodes don't curate

### ISSUE THREAD

**Issue by ordirules** [xpdAcq/SHED#45]
The following
```python
import shed.event_streams as es
s = Stream()
s2 = Stream()
s3 = es.zip(s,s2)
s.emit((name1, doc1))
s.emit((name2, doc2))
```
yields something like:
```
((name1, doc1), (name2, doc2))
```
i.e. the stream is not curated. Should we have it curated (as done by other routines) so that it becomes:
```python 
((name1, doc1, doc2))
```
or a `RuntimeError` if `name1 != name2` (as defined in `curate_streams`).

Note that the stream curation is handled automatically by the next element in the stream so this would not pose an issue for most use cases.

**Comment by CJ-Wright** [comment]
Thank you for reporting!

**Comment by CJ-Wright** [comment]
In disucssion with @ordirules: yes everything should have a call to `curate_streams` somewhere, we should be very strict that we only ever deal with `(name, (tuple of dicts))`.

**Comment by CJ-Wright** [comment]
Closed by #55
